File: src/main/scala/com/cloudant/clouseau/PerFieldAnalyzer.scala
/*
 * Copyright 2013 Cloudant. All rights reserved.
 * The Scala equivalent of PerFieldAnalyzerWrapper with a
 * public getWrappedAnalyzer method.
 */

package com.cloudant.clouseau

import org.apache.lucene.analysis._
import org.apache.lucene.analysis.Analyzer._

class PerFieldAnalyzer(defaultAnalyzer: Analyzer,
                       map: Map[String, Analyzer]) extends AnalyzerWrapper {

  def getWrappedAnalyzer(fieldName: String): Analyzer = {
    map.getOrElse(fieldName, defaultAnalyzer)
  }

  override def wrapComponents(fieldName: String,
                              components: TokenStreamComponents): TokenStreamComponents = {
    components
  }

  override def toString = {
    "PerFieldAnalyzer(" + map + ", default=" + defaultAnalyzer + ")"
  }

}

File: src/main/java/com/cloudant/cloujeau/PerFieldAnalyzer.java
package com.cloudant.cloujeau;

import java.util.Map;

import org.apache.lucene.analysis.Analyzer;
import org.apache.lucene.analysis.AnalyzerWrapper;

public class PerFieldAnalyzer extends AnalyzerWrapper {

    private final Analyzer defaultAnalyzer;
    private final Map<String, Analyzer> map;

    public PerFieldAnalyzer(final Analyzer defaultAnalyzer, final Map<String, Analyzer> map) {
        super(PER_FIELD_REUSE_STRATEGY);
        this.defaultAnalyzer = defaultAnalyzer;
        this.map = map;
    }

    @Override
    public Analyzer getWrappedAnalyzer(String fieldName) {
        return map.getOrDefault(fieldName, defaultAnalyzer);
    }

}
